tests/test_admin_views.py

```diff
@@ -0,0 +1,145 @@
+            'test_user', 'test_user@test.com', 'password')
+        self.admin_user.save()
+        self.c = Client()
+        self.c.force_login(self.admin_user)
+
+    def test_subject_delete(self):
+        """Test main delete menu is displayed."""
+        subjects = factories.SubjectFactory.create_batch(3)
+        response = self.c.get(reverse('admin_main_delete'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/delete_subject.html'
+        assert response.context['current_subject'] == ''
+        assert response.context['child_error'] is None
+        # Assert all subject objects that can be deleted are listed on html page
+        for value in subjects:
+            assert value.name.encode() in response.content
+
+    def test_subject_delete_with_sub_id(self):
+        """Test the browse structure is displayed."""
+        subject = factories.SubjectFactory.create()
+        subject_2 = factories.SubjectFactory.create(parent=subject.id)
+        response = self.c.get(reverse('admin_delete', kwargs={'sub_id': subject.id}))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/delete_subject.html'
+        assert response.context['current_subject'].name == subject.name
+        # Assert subject object's subcategories are displayed
+        assert b'Click a subject to view its subcategories.' in response.content
+        assert subject_2 in response.context['browse_list']
+        assert response.context['child_error'] is None
+
+    def test_subject_delete_sub_delete_with_subcategory(self):
+        """Test subject(s) with sub-categories/children do not get deleted."""
+        subject = factories.SubjectFactory.create()
+        factories.SubjectFactory.create(parent=subject.id)
+        response = self.c.get(reverse('admin_delete_subject',
+                                      kwargs={'sub_id': subject.id, 'sub_delete': True}))
+
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/delete_subject.html'
+        assert response.context['current_subject'].name == subject.name
+        assert response.context['browse_list'] == 'Deleted'
+        assert response.context['child_error'] == 1
+
+    def test_subject_delete_sub_delete(self):
+        """Test subject(s) with no sub-categories/children is/are deleted safely."""
+        subject = factories.SubjectFactory.create(parent=0, lft=0, rght=1)
+        response = self.c.get(reverse('admin_delete_subject',
+                                      kwargs={'sub_id': subject.id, 'sub_delete': True}))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/delete_subject.html'
+        assert response.context['current_subject'].name == subject.name
+        assert response.context['browse_list'] == 'Deleted'
+        assert response.context['child_error'] == 2
```

Comment:
I think it would be good to add comments to this line and the similar one in the previous test to indicate that a 2 for `child_error` means the subject and any child subjects were successfully deleted, while a 1 means nothing was deleted (if that's correct).

Reply:
yeah...that's correct. I added a docstring for this purpose, but I think adding a comment for `child_error` will be useful.

Reply:
The docstrings are helpful, but they don't really explain why this particular line is helpful in showing that the desired outcome was reached. Another thing that might be more obvious without needing explanation would be to check whether `subject` still exists or not at the end. In the previous test, it should still exist, but in this one, it should not